From d69bca1f0425be1a42f7ad1e3cff5313ba7c122a Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 9 Oct 2024 15:13:14 +0100
Subject: [PATCH 1305/1350] dts: bcm2712d0: Add non-d0 vc6 compatible string

Although the VC4/VC6 driver requires a special compatible string for the
"d0" stepping, the removal of the old compatible string upsets Mesa.

Satisfy both requirements by adding the old "brcm,bcm2712-vc6" string
as a fallback.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/bcm2712d0-overlay.dts   | 2 +-
 arch/arm64/boot/dts/broadcom/bcm2712d0-rpi-5-b.dts | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/arch/arm/boot/dts/overlays/bcm2712d0-overlay.dts
+++ b/arch/arm/boot/dts/overlays/bcm2712d0-overlay.dts
@@ -41,7 +41,7 @@
 	fragment@4 {
 		target = <&vc4>;
 		__overlay__ {
-			compatible = "brcm,bcm2712d0-vc6";
+			compatible = "brcm,bcm2712d0-vc6", "brcm,bcm2712-vc6";
 		};
 	};
 
--- a/arch/arm64/boot/dts/broadcom/bcm2712d0-rpi-5-b.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2712d0-rpi-5-b.dts
@@ -87,7 +87,7 @@
 };
 
 &vc4 {
-	compatible = "brcm,bcm2712d0-vc6";
+	compatible = "brcm,bcm2712d0-vc6", "brcm,bcm2712-vc6";
 };
 
 &uart10 {
